### DISCUSSION TITLE
How to set keybindings for multiple languages

### DISCUSSION THREAD

**Discussion by AmitGolden** [qtile/qtile#2453] [Q&A]
Hi, I'm a GNOME user and I have been looking into Qtile for quite some time.

I use Hebrew and English on a regular basis, but when I'm in Hebrew, my English keybinds won't work.
Is there a way to set keybindings for multiple languages, without setting each keybind twice, for every language?

Thanks

**Comment by nazar-pc** [comment]
This depends on how you switch keyboard in general. GNOME folks broke this years ago and I had to abandon gnome-settings-daemon because of that.

I set up 3 layouts with `setxkbmap` and use `gxkb` as layout indicator, `xsettingsd` can be used to set up theme, cursor and other UI goodies. Depending on your needs you can assemble working DE from pieces yourself, it just depends how deep you want to dig :slightly_smiling_face:

**Comment by ohadeytan** [comment]
Hi @AmitGolden, 
I had a similar issue and ended up putting this in my `qtile` config file (after the `keys` definition):
```python
translation = {
        'e' : 'hebrew_kuf',
        'f' : 'hebrew_kaph',
        'n' : 'hebrew_nun',
        'r' : 'hebrew_resh',
        's' : 'hebrew_dalet',
        'q' : 'slash',
        'w' : 'apostrophe',
    }

import copy
def translate_key(key):
    key = copy.copy(key)
    key.key = translation[key.key]
    return key
keys += [translate_key(key) for key in keys if key.key in translation]
``` 
using the names for the keys from [here](https://github.com/qtile/qtile/blob/master/libqtile/backend/x11/xkeysyms.py).

Not sure it's the best or easiest way, but it's working! :)
